pkg/rest/request.go

```diff
@@ -71,14 +73,26 @@ func requestWithBody(method string, url string, body io.Reader, apiToken string)
 
 func executeRequest(request *http.Request) Response {
 	client := &http.Client{}
-	resp, err := client.Do(request)
-	if err != nil {
-		println("HTTP Request failed with Error: " + err.Error())
-		return Response{}
-	}
-	defer func() {
-		err = resp.Body.Close()
-	}()
-	body, err := ioutil.ReadAll(resp.Body)
-	return Response{resp.StatusCode, body}
+
+	rateLimitStrategy := createRateLimitStrategy()
+
+	// TODO properly handle error
```

Comment:
If I read it correctly, errors inside `rateLimitStrategy` only happen if we fail to parse the expected rate limit headers? 

I find it interesting that the request callback here swallows any errors but the client.Do error (so did the method it was before) 

Are there any tests that check how the RateLimitStrategy handles a blank `Response` object from the callback?

Reply:
You are right, I'm going to return the empty response if the `rateLimitStrategy` returns an `error`.

Good catch! There are no tests for handling the empty response in `simpleSleepRateLimitStrategy`. I will add one.

Reply:
Done in 808c37c